### ISSUE TITLE
Re-discuss public API

### ISSUE THREAD

**Issue by thomkeh** [predictive-analytics-lab/EthicML#290]
I think importing stuff from EthicML is too verbose. This is the current state of the API:

```
algorithms
algorithms.inprocess
algorithms.postprocess
algorithms.preprocess
data
evaluators
metrics
preprocessing
utility
vision
visualisation
```

I think it would be nice to shorten everything a bit. For example like this:

```
algo
algo.intra
algo.post
algo.pre
data
eval
metrics
preprocess
utils
vision
plot
```

@MylesBartlett @olliethomas

**Comment by thomkeh** [comment]
Some things could even be moved to the base module directly. For example `evaluate_models` and `load_data`. Then you could write code like this:

```python
import ethicml as eml

results = eml.evaluate_models(
    datasets=[eml.data.Adult()],
    inprocess_models=[eml.algo.intra.Kamiran()],
    preprocess_models=[eml.algo.pre.Upsampler()],
    metrics=[eml.metrics.Accuracy()],
    per_sens_attribute=[eml.metrics.ProbPos()],
)
eml.plot.plot_results(results, "Accuracy", "prob_pos_Male_0/Male_1")
```

**Comment by thomkeh** [comment]
closed by #367
